File: simple-stack-example-multistack/src/main/java/com/zhuinden/simplestackdemomultistack/presentation/paths/main/chromecast/ChromeCastView.java
package com.zhuinden.simplestackdemomultistack.presentation.paths.main.chromecast;

import android.annotation.TargetApi;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.RelativeLayout;

import com.zhuinden.simplestack.Backstack;

/**
 * Created by Owner on 2017. 01. 12..
 */

public class ChromeCastView
        extends RelativeLayout {
    public ChromeCastView(Context context) {
        super(context);
        init(context);
    }

    public ChromeCastView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public ChromeCastView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    @TargetApi(21)
    public ChromeCastView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init(context);
    }

    private void init(Context context) {
        if(!isInEditMode()) {
            chromeCastKey = Backstack.getKey(context);
        }
    }

    ChromeCastKey chromeCastKey;
}

File: simple-stack-example-multistack-view/src/main/java/com/zhuinden/simplestackdemomultistack/features/main/chromecast/ChromeCastView.kt
package com.zhuinden.simplestackdemomultistack.features.main.chromecast

import android.annotation.TargetApi
import android.content.Context
import android.util.AttributeSet
import android.widget.RelativeLayout

import com.zhuinden.simplestack.Backstack

class ChromeCastView : RelativeLayout {
    lateinit var chromeCastKey: ChromeCastKey

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
    @TargetApi(21)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes)

    init {
        if (!isInEditMode) {
            chromeCastKey = Backstack.getKey(context)
        }
    }
}
